truss/cli/train_commands.py

```diff
@@ -111,14 +111,58 @@ def _prepare_click_context(f: click.Command, params: dict) -> click.Context:
     return ctx
 
 
+def _resolve_team_id(
+    remote_provider: BasetenRemote, provided_team_name: Optional[str]
+) -> Optional[str]:
+    """
+    Resolve team ID from the provided team name or by prompting the user.
+
+    Args:
+        remote_provider: The remote provider to fetch teams from
+        provided_team_name: Optional team name provided via CLI
+
+    Returns:
+        Team ID if resolved, None otherwise
+
+    Raises:
+        click.ClickException: If provided team name doesn't exist
+    """
+    existing_teams = remote_provider.api.get_teams()
+
+    # If team name provided, validate and return team ID
+    if provided_team_name is not None:
+        if provided_team_name not in existing_teams:
+            available_teams_str = remote_cli.format_available_teams(existing_teams)
+            raise click.ClickException(
```

Comment:
instead of raising an exception we could call `inquire_team` here right? and if there's only one team we could default to the default team.

Reply:
Hmm I think we should raise some sort of exception and not re-prompt. If you look at how it works if a user specifies an invalid remote, the CLI errors out and doesn't do the `inquire_remote` codepath:

```
$ truss train push config.py --remote baseten-foobar --team t3 --job-name "archit_staging_team3"
ERROR ValueError: Service provider baseten-foobar not found in ~/.trussrc
```

We should have parity with this mechanism. If the user provided option from the CLI is invalid, we should error out and not try to override. The error message is helpful enough to show available teams and so they could re-execute `truss train` with the correct team option next time.